halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/component/v1/ComponentType.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.deploy.component.v1;
+
+public enum ComponentType {
+  // These are (for now) the only services that we'll need to talk to as Halyard
+  CLOUDDRIVER;
+
+  public Class<? extends ComponentService> getServiceClass() {
+    switch (this) {
+      case CLOUDDRIVER:
+        return Clouddriver.class;
+      default:
```

Comment:
I don't see how this path can be taken.

In any case, why not just define a constructor on the enum and then just return the service class via a getter?